30_regexp_lab/relab.txt

```diff
@@ -1,17 +1,25 @@
 Use grep to print only the lines from "input" for which:
 
 1. There is a binary number in the line
+	grep -E -w '[10]+' input.txt | grep -E -v '[10]+[),]'
```

Comment:
hmm interesting idea with the 2nd grep! I  but if you only want to match binary numbers without garbage in between, I would simply test for spaces: ``` grep -E  '\s[10]+\s'``` but then:
```
vitali@Tux:~$ echo "111 crap111 c001rap 001" | grep -E  '\s[10]+\s'
``` 

to fix it we can use logical or in the regex, for example: ```(cond1 | cond2)``` will match cond1 OR cond2.. we basically want it be space padded or at the end of the line(&) or at the beginning (^):
```
vitali@Tux:~$ echo "111 crap111 c001rap 001" | grep -E  '(^|\s)[10]+($|\s)'
111 crap111 c001rap 001
```